src/components/wizards/create-vm/CreateVmWizard.js

```diff
@@ -0,0 +1,302 @@
+import { createVM } from '../../../k8s/request';
+import { FormFactory } from '../../forms/FormFactory';
+import { isPositiveNumber } from '../../../utils/validation';
+import {
+  CUSTOM_FLAVOR,
+  TEMPLATE_FLAVOR_LABEL,
+  TEMPLATE_OS_LABEL,
+  TEMPLATE_WORKLOAD_LABEL,
+  PROVISION_SOURCE_PXE,
+  PROVISION_SOURCE_URL,
+  PROVISION_SOURCE_REGISTRY
+} from '../../../constants';
+import { getTemplatesWithLabels, getTemplatesLabelValues } from '../../../utils/template';
+
+/* eslint react/no-did-update-set-state: 0 */
+export class CreateVmWizard extends React.Component {
+  state = {
+    activeStepIndex: 0,
+    basicVmSettings: {},
+    wizardValid: false,
+    namespaces: this.props.namespaces,
+    templates: this.props.templates
+  };
+
+  componentDidUpdate({ namespaces, templates }) {
+    if (namespaces !== this.props.namespaces) {
+      this.setState({
+        namespaces: this.props.namespaces
+      });
+    }
+    if (templates !== this.props.templates) {
+      this.setState({
+        templates: this.props.templates
+      });
+    }
+  }
+
+  onFormChange = (newValue, target) => {
+    let validMsg;
+
+    if (this.basicFormFields[target].validate) {
+      validMsg = this.basicFormFields[target].validate(newValue);
+    }
+    if (this.basicFormFields[target].required && newValue.trim().length === 0) {
+      validMsg = 'is required';
+    }
+
+    if (validMsg) {
+      validMsg = `${this.basicFormFields[target].title} ${validMsg}`;
+    }
+
+    const basicVmSettings = {
+      ...this.state.basicVmSettings,
+      [target]: {
+        value: newValue,
+        validMsg
+      }
+    };
+
+    this.setState(state => ({
+      basicVmSettings: {
+        ...state.basicVmSettings,
+        [target]: {
+          value: newValue,
+          validMsg
+        }
+      }
+    }));
+    this.validateWizard(basicVmSettings);
+  };
+
+  validateWizard = values => {
+    let wizardValid = true;
+
+    // check if all required fields are defined
+    const requiredKeys = Object.keys(this.basicFormFields).filter(key => this.isFieldRequired(key, values));
+    const requiredKeysInValues = Object.keys(values).filter(key => this.isFieldRequired(key, values));
+    if (requiredKeys.length !== requiredKeysInValues.length) {
+      wizardValid = false;
+    }
+
+    // check if all fields are valid
+    for (const key in values) {
+      if (
+        values[key].validMsg &&
+        (this.basicFormFields[key].isVisible ? this.basicFormFields[key].isVisible(values) : true)
+      ) {
+        wizardValid = false;
+        break;
+      }
+    }
+
+    this.setState({
+      wizardValid
+    });
+  };
+
+  getValueFromState = key => this.state[key].map(value => value.metadata.name);
+
+  isFieldRequired = (key, basicVmSettings) => {
+    const field = this.basicFormFields[key];
+    if (field.required) {
+      return field.isVisible ? field.isVisible(basicVmSettings) : true;
+    }
+    return false;
+  };
+
+  getFlavorLabel = () => {
```

Comment:
For a follow-up: improve behavior from user-perspective, so the user is still able to select any OS, na matter what pair-option (flavor,workload) has selected.

Just ideas:
- disable unselected values (to denote "There's something more")
- add sort of "reset form" button
- add generic value to the dropdows to reset selection
- etc.